barman/backup_executor.py

```diff
@@ -1859,15 +1857,12 @@ def start_backup(self, backup_info):
 
         label = "Barman backup %s %s" % (backup_info.server_name, backup_info.backup_id)
 
-        pg_version = self.postgres.server_version
-        if pg_version >= 90600:
-            # On 9.6+ execute native concurrent start backup
-            _logger.debug("Start of native concurrent backup")
-            self._concurrent_start_backup(backup_info, label)
-        else:
-            # On older Postgres use pgespresso
-            _logger.debug("Start of concurrent backup with pgespresso")
-            self._pgespresso_start_backup(backup_info, label)
+        if not self.postgres.is_minimal_postgres_version():
```

Comment:
I wonder whether we can drop this test entirely, given we will have already checked the PostgreSQL version is supported during the check which runs before a backup is executed?

I can see the rationale for having the extra checks when it couldn't be guaranteed that a supported PostgreSQL would support concurrent backup, but now that all PostgreSQL which do not support concurrent backup natively are EOL perhaps it's overkill.

Reply:
I am not sure check is called before. 
I am looking for an early place in the code to check pg_version and exit if unsupported. but I am not sure to find a unique place to perform this check.